### PR TITLE
Docs ScaffoldSplitter

### PR DESCRIPTION
**Pull request by davidRFB**
# Pull Request Template

## Description
Addition of documentation in the ScaffoldSplitter class. Some basic definition and one example of the use. 

Fix #(issue)
https://github.com/deepchem/deepchem/issues/2688#issue-991296674 

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by davidRFB** [comment]
@rbharath  ! I was working in this documentation improvement but I was wondering if it is possible or recommend to add this kind of images in the documentation to explain better some details in the splitter. 
image

image

In addition, the scaffoldSplitter needs the smiles strings in the ids field during the DiskDataset creation : 
```import deepchem as dc
data_test = ['CC(C)=CCCC(C)=CC(=O)', 'CCCC=C', 'CCCCCCCCCCCCCC', 'CC(C)Cl',
           'CCC(C)CO', 'CCCCCCCCCC(C)O', 'CCOP(=S)(OCC)SCSCC', 'CCCCCCCO',
           'CCCCCCCC(=O)OC', 'CCOP(=S)(OCC)SCSC(C)(C)C','c3ccc2nc1ccccc1cc2c3', 
           'Nc2cccc3nc1ccccc1cc23 ', 'C1CCCCCC1','OC1CCCCCC1']
Xs= np.zeros(len(data_test))
Ys = np.ones(len(data_test))
dataset = dc.data.DiskDataset.from_numpy(X=Xs,y=Ys,w=np.zeros(len(data_test)),ids=data_test)
```
this should be specified somewhere else in documentation to clarify this condition ?

**Comment by rbharath** [comment]
@davidRFB Thanks for the contribution! It looks like the flake8/doctests are failing. Can you run flake8 on the file and fix the formatting issues on the docs?

I don't think we can add images but I think adding details about the need for the smiles field would be great

**Comment by davidRFB** [comment]
Hi @rbharath  ! The flake and the doctest now work fine ! I change a little the text. There is something about the splitter. First generate groups of molecules with the same scaffold, and then sort the groups in descending order by group size. This will cause that the first group contains the biggest number of molecules with the same scaffold and the last ones contains just one molecule. In the split, the train,test and validation datasets takes the groups of molecules also in  descending order,  therefore the test and validation will contain molecules with different scaffold while the train will have a lot of molecules with repeated scaffolds. This could be a bias ?

**Comment by peastman** [comment]
Interesting question.  I don't think it's a bias: any scaffold that appears in the test set does not appear in the training set.  A more subtle question is whether this reduces the training efficiency.  Would it do a better job of extrapolating to new scaffolds if there was more diversity of scaffolds in the training set?  Possibly so, although increasing the diversity in the training set would mean decreasing diversity in the test and validation sets.  And since they're much smaller, it makes sense to put the largest scaffold sets into the training set.  Otherwise, if any single scaffold accounts for more than a few percent of the full dataset, you could get a test set that's mostly just one scaffold.

**Comment by rbharath** [comment]
@davidRFB It looks like yapf is still failing but the flake8/doctest look good! Could you rerun yapf and verify you have version `0.22` of `yapf` locally? You can check the version by running
```
yapf --version
```

**Comment by davidRFB** [comment]
Hi @rbharath  Sorry for the confusion ! I had yapf=0.30 . The change was done. However, some CI test failed. I do not know if there is something with my branch.

**Review by rbharath — COMMENTED** [review]
LGTM!

I believe the errors are due to an issue with pdbfixer/openmm version mismatch that we need to figure out.
